lib/ansible/playbook/conditional.py

```diff
@@ -212,17 +213,32 @@ def generic_visit(self, node, inside_call=False, inside_yield=False):
             # to be looking for an undefined variable, return True, otherwise fail
             try:
                 # first we extract the variable name from the error message
-                var_name = re.compile(r"'(hostvars\[.+\]|[\w_]+)' is undefined").search(str(e)).groups()[0]
+                search_var = re.compile(r"'(hostvars\[.+\]|[\w_]+)' is undefined").search(to_text(e))
```

Comment:
Style and micro-optimization note (and I know this is pre-existing code, so just don't want you to learn bad habits ;-).  regexes are relatively expensive.  Compilation isn't the most expensive part of that but they do take up some time so Python lets you compile the regex as a separate step from making use of the regex.  Whenever there's a static regex like above, I like to precompile the regex.  (On lines 40-42, you can see other regexes being pre-compiled and saved into global vars.)

Then the code here becomes something like:
``` python
search_var = HOSTVAR_UNDEFINED_REGEX.search(to_text(e))
```

(side note: I see a DEFINED_REGEX already present globally.  Should that be used here?)

Pre-compiling the regex is often a microoptimization... it shouldn't make a big difference unless it's inside of a loop or used in multiple places.  If you think that the code is more readable when the regex is closer to the code that's using it  there's a better style for using it than using re.compile().  re.search(), re.sub(), re.match, etc will all take a regular expression and the string to operate on so you don't have to call re.compile explicitly.  that would look like:
``` python
search_var = re.search(r"'(hostvars\[.+\]|[\w_]+)' is undefined", to_text(e))
```

This form is just a little shorter.  Otherwise, it does the same thing.